src/keycombination.cpp

```diff
@@ -0,0 +1,82 @@
+#include "keycombination.h"
+
+#include <locale>
+#include <tuple>
+
+namespace newsboat {
+
+KeyCombination::KeyCombination(const std::string& key, bool shift, bool control, bool alt)
+	: key(key)
+	, shift(shift)
+	, control(control)
+	, alt(alt)
+{
+}
+
+KeyCombination KeyCombination::from_bindkey(const std::string& input)
+{
+	std::string key = input;
+	bool shift = false;
+	bool control = false;
+	bool alt = false;
+
+	if (key.length() == 1 && std::isupper(key[0])) {
+		shift = true;
+		key = std::tolower(key[0]);
+	}
+	if (key.length() == 2 && key[0] == '^') {
+		control = true;
+		key = std::tolower(key[1]);
+	}
+	return KeyCombination(key, shift, control, alt);
+}
+
+std::string KeyCombination::to_bindkey_string() const
+{
+	std::string output = key;
+	if (control && output.length() == 1) {
+		output = std::string("^") + static_cast<char>(std::toupper(output[0]));
+	}
+	if (shift && output.length() == 1) {
+		output = std::toupper(output[0]);
+	}
+	return output;
```

Comment:
Personally, I'm a big fan of early returns, not sure if this is also a thing in Newsboat's code base. If not, I'd use an `else if` here, too.